src/spell/spell-info.cpp

```diff
@@ -296,25 +295,25 @@ void print_spells(PlayerType *player_ptr, SPELL_IDX target_spell, SPELL_IDX *spe
             ryakuji[4] = '\0';
         }
 
+        std::string out_val;
         if (use_menu && target_spell) {
             if (i == (target_spell - 1)) {
-                strcpy(out_val, _("  》 ", "  >  "));
+                out_val = _("  》 ", "  >  ");
             } else {
-                strcpy(out_val, "     ");
+                out_val = "     ";
             }
         } else {
-            sprintf(out_val, "  %c) ", I2A(i));
+            out_val = format("  %c) ", I2A(i));
         }
 
         if (s_ptr->slevel >= 99) {
-            strcat(out_val, format("%-30s", _("(判読不能)", "(illegible)")));
+            out_val.append(format("%-30s", _("(判読不能)", "(illegible)")));
             c_prt(TERM_L_DARK, out_val, y + i + 1, x);
             continue;
         }
 
-        const auto spell_info = exe_spell(player_ptr, use_realm, spell, SpellProcessType::INFO);
-        strcpy(info, spell_info->data());
-        concptr comment = info;
+        std::string info = exe_spell(player_ptr, use_realm, spell, SpellProcessType::INFO)->data();
+        concptr comment = info.data();
```

Comment:
exe_spell returns type `std::optional<std::string>`.
The '->' operator of std::optional calls a member of the type it holds (in this case, std::string).
Thus, the following code

```
std::string info = exe_spell(player_ptr, use_realm, spell, SpellProcessType::INFO)->data();
```

calls data() on the returned string type object, and then creates the std::string type again from the `const char*`, which is a two-step process.

To get the string type directly (which should not return std::nullopt if called with SpellProcessType::INFO), use

```
auto info = exe_spell(player_ptr, use_realm, spell, SpellProcessType::INFO).value();
```
is appropriate.

Reply:
That line is now back to what is in the current code, `const auto info = ...`, and the initialization of comment just after it has changed.